ompi/mca/osc/rdma/osc_rdma_component.c

```diff
@@ -266,7 +266,7 @@ static int ompi_osc_rdma_component_register (void)
                                             MCA_BASE_VAR_SCOPE_GROUP, &ompi_osc_rdma_btl_names);
     free(description_str);
 
-    ompi_osc_rdma_btl_alternate_names = "sm,tcp";
+    ompi_osc_rdma_btl_alternate_names = "sm,self,tcp";
```

Comment:
To provide "self" with functions such as `btl_atomic_fop.`

it's a solution to fix the following case:

```
mpirun --np 2 --mca osc rdma --mca btl tcp,self ./lock
```
where a process is calling MPI_Win_lock() on itself.  Currently the above command will segv in this case.

Reply:
I'm not very familiar with this part of the code, but the way this string is used seems to only allow the use of the BTL without verifying connectivity. For self that should be easy and should always work, it is a single flag in the ompi_proc.

Reply:
I added it based on this description of the querry_alternate_btls:

```
 * In this case an "alternate" BTL is a BTL that does not provide true RDMA but
 * can use active messages using the BTL base AM RDMA/atomics. Since more than
 * one BTL may be needed for this support the OSC component will disable the
 * use of registration-based RDMA (these BTLs will not be used) and will use
 * any remaining BTL. By default the BTLs used will be tcp and sm but any single
 * (or pair) of BTLs may be used.
 */
static int ompi_osc_rdma_query_alternate_btls (ompi_communicator_t *comm, ompi_osc_rdma_module_t *module)
{
```

self seems to fit here - so I added it to this string because otherwise it won't call the active_message init function: 
`mca_btl_base_am_rdma_init()`. I wasn't sure if a special exception for self was warrented to do anything differently - if it is I can amend it.

Reply:
puzzling, but your explanation make sense. Let's consider this resolved for now.